include/dlaf/eigensolver/reduction_to_band/impl.h

```diff
@@ -209,40 +209,36 @@ void updateTrailingPanel(const bool has_head, const std::vector<TileT<T>>& panel
   }
 }
 
-template <Backend B, typename ASender, typename WSender, typename XSender>
+template <Backend B, typename T, typename ASender, typename WSender, typename XSender>
 void hemmDiag(pika::threads::thread_priority priority, ASender&& tile_a, WSender&& tile_w,
               XSender&& tile_x) {
-  using T = dlaf::internal::SenderElementType<ASender>;
```

Comment:
IMO adding the template is an extra possible point of failure... Mismatch of types in senders generates huge compiler error messages and are hard to track, so even reducing by one the places where it can occour is a good step forward.
We realized that checking everything makes the code completely unreadable, but at least we kept the simple things...
Looking forward to use C++20 concepts... 😉

Reply:
> Looking forward to use C++20 concepts... wink

Yes, they can't come soon enough...

I changed this back to using `SenderElementType` now.